h2o-automl/src/test/java/hex/ModelTestCommons.java

```diff
@@ -0,0 +1,40 @@
+package hex;
+
+import water.H2O;
+import water.Key;
+
+import java.io.IOException;
+
+public interface ModelTestCommons {
+  public static class TestModel extends Model {
```

Comment:
@seb-h2o I see your concern but I named it this way as I was planning to put there not only stubs. It could be anything(any code snippets) that help me to test Model.
But the main reason that it is not in the subfolder is that I would have to make all stub classes and its constructors public and I don't like to pollute scope of the package(in terms of autosuggested options when you coding with IDE). In Scala for example there is an fine grained ability to control visibility of methods... you can use package-private modifiers for methods.
```
package com.acme.coolapp.model {
    class Foo {
        private[model] def doX {}
        private[coolapp] def doY {}
        private[acme] def doZ {}
    }
}
```
so we could do  private[hex] and put it into hex.stub package.

If you know how to put this ModelTestCommons class to `hex.stub` package without making it `all public` please suggest me.
Regarding `no one will ever look into something` - this is a subjective thing. There is no such a package with name 'stub' in our whole project so it means there is no such a convention yet so that I should follow it. I would be really happy if we had a way to efficiently share and keep all our conventions.

Reply:
@deil87 there's no subjectivity here: when I search for a recipe with broccoli, I type "recipe with broccoli",  for test, I type "test", when searching stubs, I type "stub", not `commons`... and I believe (you may say it's just subjective) that the great majority of people does the same.
There's no package named `stub` today, because I bet there's almost no stub written yet: it's up to you to create one. package/class conventions in Java are the same as in Scala: name things according to what they contain or represent, so, as a general rule, `commons` is good name when you want to put anything... which is the definition of a bad name.

To answer your concern about visibility, we're talking about tests package here: you may have noticed that all tests are public, I don't understand your concern about keeping stub classes package-private.

Reply:
@seb-h2o Regarding visibility, we don't have many tests yet that is why it is not a concern yet. I was working on the projects where  even test package was quite full of classes as ideally we should have at least one test class for corresponding class in main package. And even more given the fact that there is always a lot of extra stub or fake entities around tests. And I'm used to care about developer's comfort when he is writing tests. 
Regarding `common`...say it to apache commons :)  I was planning to have `commons` but now it looks definitely like a stub.  
Overall I agree with you and I will move and rename.

Reply:
apache-commons is the worst name Apache ever came with :)

Reply:
@seb-h2o I can't move stubs to `hex.stub` as I will have to change access modifier in `Model.java` for `modelDescriptor()` method from package-private to public. `ModelStub` class should be in the same package as `Model` class. So I'm going to just rename to `ModelStub`.